sqlmesh/core/model/kind.py

```diff
@@ -404,10 +431,14 @@ def _model_kind_validator(cls: t.Type, v: t.Any, values: t.Dict[str, t.Any]) ->
         if time_data_type:
             props["time_data_type"] = exp.DataType.build(time_data_type, dialect=dialect)
         name = v.this if isinstance(v, d.ModelKind) else props.get("name")
+        # We want to ensure whatever name is provided to construct the class is the same name that will be
```

Comment:
this comment explains what, but i'm not sure it explains why? what's the reasoning behind this

Reply:
I was thinking the example explained why: 
```
Ex: Pass in `SCD_TYPE_2` then we want to ensure we get `SCD_TYPE_2` as the kind name instead of `SCD_TYPE_2_BY_TIME`.
```
In other words, for legacy aliases we want to ensure we maintain the legacy name instead of using the new name which is what would happen if we relied on the default value.

Reply:
yea why though? why can't SCD_TYPE_2 map to the new name? why do we need to keep the old mapping?

Reply:
Oh to avoid having it be detected as a change to the model when you run plan/apply. Will add that note.